zkevm-circuits/src/evm_circuit/execution.rs

```diff
@@ -723,33 +748,52 @@ impl<F: Field> ExecutionConfig<F> {
 
                 self.q_step_first.enable(&mut region, offset)?;
 
-                // handle EndBlock
-                let dummy_tx = Transaction {
-                    calls: vec![Default::default()],
-                    ..Default::default()
-                };
-                let last_tx = block.txs.last().unwrap_or(&dummy_tx);
-                let end_block_state = &ExecStep {
-                    rw_counter: if block.txs.is_empty() {
-                        0
-                    } else {
-                        // if it is the first tx,  less 1 rw lookup, refer to end_tx gadget
-                        last_tx.steps.last().unwrap().rw_counter + 9 - (last_tx.id == 1) as usize
-                    },
-                    execution_state: ExecutionState::EndBlock,
-                    ..Default::default()
-                };
+                let dummy_tx = Transaction::default();
```

Comment:
We defined a dummy() for Transaction above but we use Transaction::default here, is this intended?

Reply:
Ah, that `Transaction::dummy()` is for the `circuit_input_builder::Transaction`, whereas this one is `witness::Transaction`.

I created a `dummy` function for `circuit_input_builder::Transaction` to show the intention that it's a dummy tx that contains dummy data, instead of using a default which to me seems like it could create a valid default tx.

Reply:
Note: we have a pending task which is trying to unify the types between `circuit_input_builder` and `witness` modules (which we currently have duplicated).  See https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/528